synapse/lib/view.py

```diff
@@ -289,13 +289,17 @@ def isForkOf(self, viewiden):
     async def _calcForkLayers(self):
         # recompute the proper set of layers for a forked view
         # (this may only be called from within a nexus handler)
-        layers = [self.layers[0]]
+        layers = []
 
-        view = self.parent
-        while view is not None:
+        # Add one layer from each of the views that aren't the bottom view
+        view = self
+        while view.parent is not None:
             layers.append(view.layers[0])
             view = view.parent
 
+        # Add all of the bottom view's layers
```

Comment:
It feels odd to handle only the last view with a special case here. We may need to talk this one through just one more time :D